src/Symfony/Bundle/FrameworkBundle/CacheWarmer/TranslationsCacheWarmer.php

```diff
@@ -22,18 +23,34 @@
  */
 class TranslationsCacheWarmer implements CacheWarmerInterface
 {
+    private $container;
     private $translator;
 
-    public function __construct(TranslatorInterface $translator)
+    public function __construct($container)
     {
-        $this->translator = $translator;
+        /*
+         * This is an optional cache warmer.
+         * It means this cache is not warm on boot but on explicit cache warm.
+         * To avoid loading the translator service on boot, we inject the container here.
+         */
```

Comment:
What about:

As this cache warmer is optional, dependencies should be lazy-loaded, that's why a container should be injected.